### PR TITLE
more manual stuff 2

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: gdk4/src/auto/clipboard.rs [hunk 1]
```diff
@@ -249,13 +244,8 @@ impl Clipboard {
...
     }
 
-    //#[doc(alias = "gdk_clipboard_set_valist")]
-    //pub fn set_valist(&self, type_: glib::types::Type, args: /*Unknown conversion*//*Unimplemented*/Unsupported) {
-    //    unsafe { TODO: call ffi:gdk_clipboard_set_valist() }
-    //}
-
     #[doc(alias = "gdk_clipboard_set_value")]
-    pub fn set_value(&self, value: &glib::Value) {
+    pub fn set(&self, value: &glib::Value) {
```
Comment (sdroege) [inline gdk4/src/auto/clipboard.rs:248]: Why not `fn set(&self, value: &dyn ToValue)`?

File: gdk4/src/display.rs [hunk 2]
```diff
@@ -39,6 +39,30 @@ impl Display {
         }
     }
 
+    #[doc(alias = "gdk_display_map_keycode")]
+    pub fn map_keycode(&self, keycode: u32) -> Option<(Vec<KeymapKey>, Vec<Key>)> {
```
Comment (sdroege) [inline gdk4/src/display.rs:43]:
```suggestion
    pub fn map_keycode(&self, keycode: u32) -> Option<Vec<(KeymapKey, Key)>> {
```

Maybe?

File: gdk4/src/auto/clipboard.rs (hunk 1 above)
Comment (sdroege) [reply gdk4/src/auto/clipboard.rs:248]: Maybe the same elsewhere

Comment (bilelmoussaoui) [reply gdk4/src/auto/clipboard.rs:248]: Yes, that's much nicer

File: gdk4/src/display.rs (hunk 2 above)
Comment (bilelmoussaoui) [reply gdk4/src/display.rs]: Thanks!

File: gdk4/src/texture.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+use glib::translate::*;
+use glib::IsA;
+
+pub trait TextureExtManual: 'static {
+    #[doc(alias = "gdk_texture_download")]
+    fn download(&self, stride: usize) -> Vec<u8>;
+}
+
+impl<O: IsA<Texture>> TextureExtManual for O {
+    fn download(&self, stride: usize) -> Vec<u8> {
```
Comment (bilelmoussaoui) [inline gdk4/src/texture.rs:13]: Not sure if this is nicer than having to pass to a `mut [u8]`?

Comment (sdroege) [reply gdk4/src/texture.rs:13]: Yes, `&mut [u8]` here and `&[u8]` for upload

**Pull request merged**